decoder/links.go

```diff
@@ -56,11 +75,16 @@ func (d *PathDecoder) linksInBody(body *hclsyntax.Body, bodySchema *schema.BodyS
 					}
 				}
```

Comment:
I didn't think of this before when we discussed this earlier but could we not range here over [`dk.Attributes`](https://github.com/hashicorp/hcl-lang/blob/f24c834260e015170a294693b9971e4ff1c219ab/schema/dependent_schema.go#L19) here, instead of merging dependent schemas below? 🤔 

That would also avoid the need to deal with attributes above.

Reply:
I missed that `dk` also has an attributes field. Ranging over this makes things much easier, thanks!